torch/distributions/weibull.py

```diff
@@ -0,0 +1,52 @@
+from numbers import Number
+import math
+import torch
+from torch.distributions import constraints
+from torch.distributions.uniform import Uniform
+from torch.distributions.transformed_distribution import TransformedDistribution
+from torch.distributions.transforms import AffineTransform, ExpTransform, PowerTransform
+from torch.distributions.utils import _finfo, broadcast_all
+from torch.distributions.gumbel import euler_constant
+
+
+class Weibull(TransformedDistribution):
+    r"""
+    Samples from a two-parameter Weibull distribution.
+
+    Example:
+
+        >>> m = Weibull(torch.tensor([1.0]), torch.tensor([1.0]))
+        >>> m.sample()  # sample from a Weibull distribution with scale=1, shape=1
+        tensor([ 0.4784])
+
+    Args:
+        scale (float or Tensor): Scale parameter of distribution (lambda).
+        shape (float or Tensor): Shape parameter of distribution (k).
+    """
+    arg_constraints = {'scale': constraints.positive, 'shape': constraints.positive}
+    support = constraints.positive
+
+    def __init__(self, scale, shape, validate_args=None):
+        self.scale, self.shape = broadcast_all(scale, shape)
+        finfo = _finfo(self.scale)
+        if isinstance(scale, Number) and isinstance(shape, Number):
+            batch_shape = torch.Size()
+            base_dist = Uniform(finfo.tiny, 1 - finfo.eps)
+        else:
+            batch_shape = self.scale.size()
+            base_dist = Uniform(self.scale.new(self.scale.size()).fill_(finfo.tiny), 1 - finfo.eps)
+        transforms = [ExpTransform().inv, AffineTransform(loc=0, scale=-1),
+                      PowerTransform(exponent=self.shape.reciprocal()), AffineTransform(loc=0, scale=self.scale)]
```

Comment:
Since `self.shape.reciprocal()` appears to be used in every method, even in `__init__()`, I think it makes sense to define it as a property. WDYT?
```py
self.scale, self.shape = broadcast_all(scale, shape)
self.shape_reciprocal = self.shape.reciprocal()
```